File: 485. Max Consecutive Ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] a) {
        int max = 0;
        int currMax = 0;
        for (int i =0; i<a.length;i++) {
            if (a[i]==1) {
                currMax++;
            } else {
                max = Math.max(max, currMax);
                currMax = 0;
            }
        }
        return Math.max(max, currMax);
    }
}

File: python/485. Max Consecutive Ones.py
# 485. Max Consecutive Ones
# https://leetcode.com/problems/max-consecutive-ones/
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        mx, curr = 0, 0
        for num in nums:
            if num == 0:
                mx = max(mx, curr)
                curr = 0
            else:
                curr += 1
        return max(mx, curr)
